internal/xds/translator/translator.go

```diff
@@ -509,10 +511,31 @@ func processXdsCluster(tCtx *types.ResourceVersionTable, httpRoute *ir.HTTPRoute
 		endpointType = EndpointTypeStatic
 	}
 
+	tsocket := &corev3.TransportSocket{}
+	tsocket = nil
+
+	ca := mergeCaBytes(httpRoute.Destination.Settings)
+
+	if string(ca) != "" {
+		// use destination name as the ca secret name
+		caXdsSecretName := httpRoute.Destination.Name
+		CaSecret := buildXdsUpstreamTLSCASecret(caXdsSecretName, ca)
+		if err := tCtx.AddXdsResource(resourcev3.SecretType, CaSecret); err != nil {
+			return err
+		}
+		// for upstreamTLS , a fixed sni can be used. use auto_sni otherwise
+		// https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/sni#faq-how-to-setup-sni:~:text=For%20clusters%2C%20a,for%20trust%20anchor.
+		temp, err := buildXdsUpstreamTLSSocketWthCert(caXdsSecretName, httpRoute.Hostname)
```

Comment:
this doesnt look right, Id suggest only creating one tls socket, using all the info from `destintion.settings[0].tls`
as a follow up we can fix this the right way which is creating multiple tls sockets
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-cluster-transportsocketmatch

Reply:
fixed